core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -638,6 +666,24 @@ static Parameter ofRequestObject(Class<?> type,
             this.name = name;
             this.defaultValue = defaultValue;
             this.requestConverterFunction = requestConverterFunction;
+
+            if (type.isEnum()) {
+                final Set<? extends Enum<?>> enumInstances = EnumSet.allOf(type.asSubclass(Enum.class));
+                upperCaseEnum = enumInstances.stream().collect(toImmutableMap(e -> e.name().toUpperCase(),
+                                                                              Function.identity()));
+                isEnum = true;
+            } else {
+                upperCaseEnum = null;
+                isEnum = false;
+            }
+        }
+
+        Map<String, Enum<?>> getUpperCaseEnum() {
```

Comment:
How about removing this method and instead:
```
@SuppressWarnings("unchecked")
<T extends Enum<T>> T get(String key) {
    return (T) upperCaseEnum.get(Ascii.toUpperCase(key));
}
```

Reply:
Checked & Updated Thanks